src/ambianic/pipeline/ai/fall_detect.py

```diff
@@ -72,39 +72,63 @@ def calculate_angle(self, p):
             
         theta = abs(theta1-theta2)
         return theta
+    
+    def find_keypoints(self, image):
+
+        image_ori = image.copy()
+        rotation = [90, -90]
+        
+        while True:
+            poses = self._pose_engine.DetectPosesInImage(image)
+            for pose in poses:
```

Comment:
@bhavikapanara is there a chance that poses is None  or empty? In such cases this while True loop will spin forever, no?

Reply:
No...there are no chances to get an empty result.
that will always return key-points. If there is no person in a frame, it will return key-points with very very low probability score.

I have checked for this image [no person](https://github.com/ambianic/ambianic-edge/blob/fall-detect/tests/pipeline/ai/background.jpg) image. below is the return response from posenet.


`[Pose({'nose': Keypoint(<nose>, [1335, 171], 0.0018767025657813801), 'left eye': Keypoint(<left eye>, [1295, 6], 0.0011504802784690327), 'right eye': Keypoint(<right eye>, [1305, 11], 0.0015948671443740245), 'left ear': Keypoint(<left ear>, [324, 445], 0.003422743024364482), 'right ear': Keypoint(<right ear>, [478, 347], 0.0011285191402948904), 'left shoulder': Keypoint(<left shoulder>, [1160, 499], 0.006819241507138113), 'right shoulder': Keypoint(<right shoulder>, [1116, 487], 0.0022483134904298474), 'left elbow': Keypoint(<left elbow>, [672, 754], 0.011247683477665103), 'right elbow': Keypoint(<right elbow>, [1071, 552], 0.0051987406127019156), 'left wrist': Keypoint(<left wrist>, [588, 734], 0.013236690572867188), 'right wrist': Keypoint(<right wrist>, [403, 653], 0.008069606264386485), 'left hip': Keypoint(<left hip>, [1111, 740], 0.0112793711392602), 'right hip': Keypoint(<right hip>, [1066, 737], 0.01047226251029573), 'left knee': Keypoint(<left knee>, [638, 496], 0.0036192633880051815), 'right knee': Keypoint(<right knee>, [468, 521], 0.0037731794967397485), 'left ankle': Keypoint(<left ankle>, [488, 630], 0.03046229198464312), 'right ankle': Keypoint(<right ankle>, [473, 628], 0.03632458557985564)}, 0.0)]
`

Here, in the response, last value **`0.0`** is the pose score, it depends on a number of key-points are detected with more than 60% probability score.

Reply:
That is good. As a best practice, it's good to account for edge cases that may put your code in an infinite loop. I would still add a check to exit the loop in case there are no poses returned. You never know if the model has a bug or the next version of the model trims results below a given confidence threshold of top-k threshold. Either way infinite blocking  is not desirable. Better to have some sort of exit - either positive or an error report.

Reply:
Yes...sure. I think it is best practise to consider all scenario. Sorry, I forgot it.
Will update the code

Reply:
Great. Thank you!

Reply:
Thanks for the fix. Closing.